client/my-sites/people/team-list/team.jsx

```diff
@@ -0,0 +1,146 @@
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			bulkEditing: false
+		};
+	}
+
+	render() {
+		const { props, state } = this;
+		const { translate } = props;
+		const key = deterministicStringify( omit( props.fetchOptions, [ 'number', 'offset' ] ) );
+		const	listClass = classNames( {
+			'bulk-editing': state.bulkEditing,
+		} );
+
+		let headerText = translate( 'Team', { context: 'A navigation label.' } );
+		let people;
+
+		const isLastPage = () => props.totalUsers <= props.users.length + props.excludedUsers.length;
+		const renderPerson = ( user ) => {
+			return (
+				<PeopleListItem
+					key={ user.ID }
+					user={ user }
+					type="user"
+					site={ props.site }
+					isSelectable={ state.bulkEditing } />
+			);
+		};
+
+		const fetchNextPage = () => {
+			const offset = props.users.length;
+			const fetchOptions = Object.assign( {}, props.fetchOptions, { offset: offset } );
+			analytics.ga.recordEvent( 'People', 'Fetched more users with infinite list', 'offset', offset );
+			debug( 'fetching next batch of users' );
+			UsersActions.fetchUsers( fetchOptions );
+		};
+
+		const getPersonRef = ( user ) => `user-${ user.ID }`;
+
+		const renderLoadingPeople = () => <PeopleListItem key="people-list-item-placeholder" />;
+
+		if ( props.fetchInitialized && ! props.users.length && props.fetchOptions.search && ! props.fetchingUsers ) {
+			return (
+				<NoResults
+					image="/calypso/images/people/mystery-person.svg"
+					text={
+						translate( 'No results found for {{em}}%(searchTerm)s{{/em}}',
+							{
+								args: { searchTerm: props.search },
+								components: { em: <em /> }
+							}
+						)
+					} />
+			);
+		}
+
+		if ( props.site && props.users.length ) {
+			if ( props.search && props.totalUsers ) {
+				headerText = translate(
+					'%(numberPeople)d Person Matching {{em}}"%(searchTerm)s"{{/em}}',
+					'%(numberPeople)d People Matching {{em}}"%(searchTerm)s"{{/em}}',
+					{
+						count: props.users.length,
+						args: {
+							numberPeople: props.totalUsers,
+							searchTerm: props.search
+						},
+						components: {
+							em: <em />
+						}
+					}
+				);
+			}
+
+			people = (
+				<InfiniteList
+					key={ key }
+					items={ props.users }
+					className="people-selector__infinite-list"
+					ref="infiniteList"
+					fetchingNextPage={ props.fetchingUsers }
+					lastPage={ isLastPage() }
+					fetchNextPage={ fetchNextPage }
+					getItemRef={ getPersonRef }
+					renderLoadingPlaceholders={ renderLoadingPeople }
+					renderItem={ renderPerson }
+					guessedItemHeight={ 126 }>
+				</InfiniteList>
+			);
+		} else {
+			people = renderLoadingPeople();
+		}
+
+		return (
+			<div>
+				<PeopleListSectionHeader
+					label={ headerText }
+					site={ props.site }
+					count={ props.fetchingUsers || props.fetchOptions.search ? null : props.totalUsers } />
+				<Card className={ listClass }>
+					{ people }
```

Comment:
I like it :) reminds me of `cond` in clojure.  Seems like a useful general function to place somewhere.  It might get expensive to actually evaluate `React.createElement` for all components in each pair regardless of whether they are used so perhaps they should instead be a function that returns a component.

The logic in this render method is pretty hard to follow - I made it worse in the role filtering PR so am keen to simplify it in this hygeine PR so that it's easier to deal with other filtering conditions.